vdj_ann/src/annotate.rs

```diff
@@ -1530,31 +1528,24 @@ pub fn annotate_seq_core(
                 let t = annx[data[i1].2[j]].ref_id;
                 if utr2 || !refdata.is_u(t as usize) {
                     let x = &data[i1].0[j];
-                    for m in x.0..x.1 {
-                        cov1[m] = true;
+                    for c in &mut cov1[x.0..x.1] {
+                        *c = true;
                     }
                 }
             }
             for j in 0..data[i2].0.len() {
                 let t = annx[data[i2].2[j]].ref_id;
                 if utr1 || !refdata.is_u(t as usize) {
                     let x = &data[i2].0[j];
-                    for m in x.0..x.1 {
-                        cov2[m] = true;
+                    for c in &mut cov2[x.0..x.1] {
+                        *c = true;
                     }
                 }
             }
-            let (mut total1, mut total2) = (0, 0);
-            for l in 0..n {
-                if cov1[l] {
-                    total1 += 1;
-                }
-            }
-            for l in 0..n {
-                if cov2[l] {
-                    total2 += 1;
-                }
-            }
+            let count_true = |bools: &[bool]| (bools[0..n]).iter().filter(|c| **c).count();
+
+            let total1 = count_true(&cov1);
+            let total2 = count_true(&cov2);
```

Comment:
A bit more easier to understand?
```suggestion
            let count_true = |bools: &[bool]| bools.iter().filter(|c| **c).count();

            let total1 = count_true(&cov1[0..n]);
            let total2 = count_true(&cov2[0..n]);
```